crates/volta-core/src/session.rs

```diff
@@ -428,10 +229,7 @@ impl Session {
 
 fn publish_plugin(hooks: &LazyHookConfig) -> Fallible<Option<&Publish>> {
     let hooks = hooks.get()?;
-    Ok(hooks
-        .events
-        .as_ref()
-        .and_then(|events| events.publish.as_ref()))
+    Ok(hooks.events().and_then(|events| events.publish.as_ref()))
```

Comment:
One note: this whole body can become:

```rust
hooks.get().map(|hooks| hooks.events().and_then(|events| events.publish.as_ref()))
```

This isn't a necessary refactor; I just tend to lean that direction when I see a `?` immediately followed by an `Ok` wrapping something.

Reply:
While I like the functional syntax and the `map` / `and_then` methods, I tend to avoid nesting them, as it starts to feel like "callback hell" and I find it a little difficult to parse. Using the `?` here lets us be nearly as brief, while only having one level of lambda.

Reply:
Yeah, that’s totally fair. One thought further, then—what are your thoughts on this?

```rust
let  = hooks.get()?;
let events = hooks.events().and_then(|events| events.publish.as_ref());
Ok(events)
```

(Again, no particular pressure to change it; that’s just noting what reads more clearly to me.)

Reply:
That's reasonable, I'll make that change :+1: